src/OpenLoco/Windows/PromptBrowseWindow.cpp

```diff
@@ -928,20 +929,23 @@ namespace OpenLoco::Ui::PromptBrowse
 
             // Copy directory and filename to buffer.
             strncpy(&_directory[0], path.u8string().c_str(), std::size(_directory));
-
-            // Close browse window to continue saving.
-            WindowManager::close(self);
         }
         else
         {
             // Copy scenario path into expected address. Trailing / on directory is assumed.
             // TODO: refactor to fs::path?
             strncat(_directory, inputSession.buffer.c_str(), std::size(_directory));
             strncat(_directory, getExtensionFromFileType(_fileType), std::size(_directory));
-
-            // Close browse window to start loading.
-            WindowManager::close(self);
         }
+
+        // Save last directory
+        fs::path lastDirectory = fs::u8path(&_directory[0]);
+        Config::getNew().last_save_path = lastDirectory.u8string();
```

Comment:
I overlooked something. Despite the name `_directory`, this contains the full path to the save game that is being loaded. Perhaps the easiest solution would be to check `lastDirectory.is_directory` and use `lastDirectory.parent_path` if it is not.